instrumentation/micrometer/micrometer-1.5/library/src/main/java/io/opentelemetry/instrumentation/micrometer/v1_5/LongMeasurementRecorder.java

```diff
@@ -32,7 +32,8 @@ final class LongMeasurementRecorder<T> implements Consumer<ObservableLongMeasure
   public void accept(ObservableLongMeasurement measurement) {
     T obj = objWeakRef.get();
     if (obj != null) {
-      measurement.record(metricFunction.applyAsLong(obj), attributes);
+      MeasurementRecorderUtil.runInThreadContextClassLoader(metricFunction.getClass().getClassLoader(),
```

Comment:
If anything, we should probably use `obj`'s class instead -- since this is the actual object being measured.
```suggestion
      MeasurementRecorderUtil.runInThreadContextClassLoader(obj.getClass().getClassLoader(),
```